src/devices/Mcp3008/Mcp3008.cs

```diff
@@ -56,88 +60,102 @@ public void Dispose()
             }
         }
 
-        public int Read(int adc_channel)
+        public int Read(int channel, InputConfiguration inputConfiguration = InputConfiguration.SingleEnded)
         {
-            if (adc_channel < 0 || adc_channel > 7)
+            if (channel < 0 || channel > 7)
             {
                 throw new ArgumentException("ADC channel must be within 0-7 range.");
             }
 
             if (_protocol == CommunicationProtocol.Spi)
             {
-                return ReadSpi(adc_channel);
+                return ReadSpi(channel, inputConfiguration);
             }
             else
             {
-                return ReadGpio(adc_channel);
+                return ReadGpio(channel, inputConfiguration);
             }
         }
 
+        private static byte GetConfigurationBits(int channel,InputConfiguration inputConfiguration = InputConfiguration.SingleEnded)
```

Comment:
NIT: space after the comma on the parameters.

Reply:
@joperezr Good catch.  I also plan to remove the optional argument value from this method as it is only called from the other private Read methods.